lib/spack/spack/installer.py

```diff
@@ -2355,9 +2355,14 @@ def get_deptypes(self, pkg):
         Returns:
             tuple: required dependency type(s) for the package
         """
+        cache_only = self.install_args.get("package_cache_only")
+        # This is likely never true, but check it to be safe
```

Comment:
Not now, but in the future this might change. Please use the unique ID which is `pkg_id`, which depends on dag_hash.

Reply:
ah, noted. I don't know if it will ever be possible for root packages to have duplicates in a build tree though. Can you think of a case, or is this more a "best practices" kind of thing?

Reply:
In the future we may have cross compilation, so then deptype=build targets the host, deptype=link targets the ... target, which causes duplication.

And even sooner we may relax "one spec per package" to "one spec per package per link-type connected subgraph"